src/query/api/v1/handler/prometheus/native/read.go

```diff
@@ -97,9 +98,14 @@ func (h *PromReadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if strings.ToLower(r.Header.Get("X-M3-Render-Format")) == "m3ql" {
```

Comment:
instead of this, can you add Format to RequestParams and then parse this header at the same place other things are parsed ? This would prevent us from adding adhoc parsing in our handlers.